adds broadcasting test cases

ryanneph opened the pull request:
refs: #80, #90, #104, #105

adamritter commented:
Is the ((5,13,24,16), (5,1,24,1)) case really important in practice? If we decide implement only 1s on the start or end (which is simpler to implement without complicating the kernel), the test should already contain only these easy cases

ryanneph commented:
> Is the ((5,13,24,16), (5,1,24,1)) case really important in practice? If we decide implement only 1s on the start or end (which is simpler to implement without complicating the kernel), the test should already contain only these easy cases

Thanks for the review. No it’s probably not important in practice. I added a few cases that I thought would test for generality. If the consensus is that we don’t want/need general broadcasting on GPU, the test cases can be modified. If that decision is made, we will be fragmenting the cpu and GPU capabilities of the library. 

Either way, this PR is designed to get make more broadcasting tests available. The content of the tests can and should be discussed, and modified, just like we are doing now — so thanks again for weighing in.

adamritter commented:
Please take a look at https://github.com/geohot/tinygrad/pull/108/ . With
it the important GPU test cases should be covered probably (I'm not sure
though).

On Thu, Nov 12, 2020 at 11:30 AM Ryan Neph <notifications@github.com> wrote:

> Is the ((5,13,24,16), (5,1,24,1)) case really important in practice? If we
> decide implement only 1s on the start or end (which is simpler to implement
> without complicating the kernel), the test should already contain only
> these easy cases
>
> Thanks for the review. No it’s probably not important in practice. I added
> a few cases that I thought would test for generality. If the consensus is
> that we don’t want/need general broadcasting on GPU, the test cases can be
> modified. If that decision is made, we will be fragmenting the cpu and GPU
> capabilities of the library.
>
> Either way, this PR is designed to get make more broadcasting tests
> available. The content of the tests can and should be discussed, and
> modified, just like we are doing now — so thanks again for weighing in.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/geohot/tinygrad/pull/106#issuecomment-726021051>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AN5SWAFCTRISAVGDZU653FLSPPBMRANCNFSM4TS53ZKQ>
> .
>
